cargo-miri/bin.rs

```diff
@@ -646,14 +737,15 @@ fn phase_cargo_rustc(args: env::Args) {
     }
 }
 
-fn phase_cargo_runner(binary: &Path, binary_args: env::Args) {
+fn phase_cargo_runner(binary: &Path, mut binary_args: env::Args) {
     let verbose = std::env::var_os("MIRI_VERBOSE").is_some();
 
     let file = File::open(&binary)
         .unwrap_or_else(|_| show_error(format!("file {:?} not found or `cargo-miri` invoked incorrectly; please only invoke this binary through `cargo miri`", binary)));
     let file = BufReader::new(file);
-    let info: CrateRunInfo = serde_json::from_reader(file)
-        .unwrap_or_else(|_| show_error(format!("file {:?} contains outdated or invalid JSON; try `cargo clean`", binary)));
+    let info: CrateRunInfo = serde_json::from_reader(file).unwrap_or_else(|_| {
+        tail_exec(Command::new(binary).args(&mut binary_args))
```

Comment:
This workaround from #1675 is kept because I can't find a good way to stop Cargo from using cargo-miri to execute the binary or disabling unit testing.